src/Ledger/Foreign/HSLedger.agda

```diff
@@ -93,50 +93,57 @@ HSScriptStructure = record { p1s = HSP1ScriptStructure ; ps = HSP2ScriptStructur
 
 open import Data.Nat
 open import Data.Nat.Properties using (+-0-commutativeMonoid; _≟_)
-open import Ledger.TokenAlgebra
+open import Ledger.TokenAlgebra.Base using (TokenAlgebra)
 open import Ledger.Transaction
 
-coinTokenAlgebra : TokenAlgebra
-coinTokenAlgebra = record
-  { PolicyId                = ℕ
-  ; Value-CommutativeMonoid = +-0-commutativeMonoid
-  ; coin                    = id
-  ; inject                  = id
-  ; policies                = λ x → ∅
-  ; size                    = λ x → 1 -- there is only ada in this token algebra
-  ; property                = λ x → refl
-  ; coinIsMonoidMorphism    = record
-    { mn-homo = record
-      { sm-homo = record { ⟦⟧-cong = λ z → z ; ∙-homo  = λ x y → refl }
-      ; ε-homo  = refl
-      }
-    }
-  ; _≤ᵗ_                    = _≤_
-  ; DecEq-Value             = record { _≟_ = Data.Nat._≟_ }
-  }
+
+module _ where
+  open TokenAlgebra
+  open IsCommutativeMonoidMorphism
+  coinTokenAlgebra : TokenAlgebra
+  coinTokenAlgebra .Value-CommutativeMonoid  = +-0-commutativeMonoid
+  coinTokenAlgebra .coin = id
+  coinTokenAlgebra .inject = id
+  coinTokenAlgebra .size = λ _ → 1
+  coinTokenAlgebra ._≤ᵗ_ = _≤_
+  coinTokenAlgebra .AssetNameType = ℕ
+  coinTokenAlgebra .property = λ _ → refl
+  coinTokenAlgebra .coinIsMonoidMorphism .mn-homo =
+    -- record
```

Comment:
We should remove the unnecessary parts (which were accidentally left commented out instead of removed).